etcdserver/server.go

```diff
@@ -385,6 +374,23 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 				plog.Panicf("recovered store from snapshot error: %v", err)
 			}
 			plog.Infof("recovered store from snapshot at index %d", snapshot.Metadata.Index)
+
+			// if follower recieves snapshot from leader and crashes before renaming xxx.snap.db to db,
+			// restarting follower results loading a outdated db.
+			// In this case:
+			// 1. check if xxx.snap.db (xxx==snapshot.Metadata.Index) exists.
+			// 2. rename xxx.snap.db to db if exists.
+			// 3. load backend again with the new db file.
+			snapfn, err := snap.GetDBFilePathByID(cfg.SnapDir(), snapshot.Metadata.Index)
```

Comment:
to use snap.DBFilePath() function? snap.DBFilePath() is tied to an object. that's why I created another GetDBFilePathByID() that's a class function.